src/cmd.cpp

```diff
@@ -140,40 +155,52 @@ static int subcommand_dem2tintiles(bool need_help,
 
     const std::string meshing_method = local_varmap["method"].as<std::string>();
 
-    auto input_raster = std::make_unique<RasterDouble>();
-
-    if(!load_raster_file(input_file.c_str(), *input_raster))
+    if("zemlya" != meshing_method && "terra" != meshing_method)
     {
-        return false;
+        throw po::error(std::string("unknown method ") + meshing_method);
     }
 
-    if("zemlya" == meshing_method || "terra" == meshing_method)
+    std::unique_ptr<char[]> emergency_memory(new char[16384]);
+
+    try
```

Comment:
wonder if this has performance cost in our case https://stackoverflow.com/questions/1897940/in-what-ways-do-c-exceptions-slow-down-code-when-there-are-no-exceptions-thown